datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -364,6 +368,42 @@ impl LogicalPlan {
     ) -> Result<LogicalPlan, DataFusionError> {
         from_plan(self, &self.expressions(), inputs)
     }
+
+    /// Convert a prepare logical plan into its inner logical plan with all params replaced with their corresponding values
+    pub fn execute(
```

Comment:
While the `execute` name matches how this function will likely be used, I think naming it something more specific to what it is doing might be clearer.

What about something like 

```suggestion
    pub fn with_param_values(self, param_values: Vec<ScalarValue>)
```

That way to use this one could do

```
let plan = logical_plan
  .with_param_values(param_value)
```